lib/influx.rb

```diff
@@ -45,43 +45,89 @@ def insert_incident(incident)
       @influxdb.write_point(timeseries, incident)
     end
 
-    def incidents(date=nil)
+    def find_incidents(start_date=nil, end_date=nil, query_input=nil)
       timeseries = @config['series']
-      unless date.nil?
-        start_date = Chronic.parse(date, :guess => false).first
-        end_date = Chronic.parse(date, :guess => false).last
-      end
+      start_date = Chronic.parse(start_date, :guess => false).first unless start_date.nil?
+      end_date = Chronic.parse(end_date, :guess => false).last unless end_date.nil?
 
       # If we couldn't parse the given date, use the last 24 hours.
       end_date = end_date.nil? ? Time.now.to_i : end_date.to_i
       start_date = start_date.nil? ? end_date - (24 * 60 * 60) : start_date.to_i
 
-      query = "select * from #{timeseries} where time > #{start_date.to_i}s and time < #{end_date.to_i}s"
-      incidents = @influxdb.query(query)
+      # As a default, select * from the timeframe.  Otherwise, use what the input query gave us
+      query_select = 'select *'
+      if query_input && query_input[:query_select]
+        query_select = query_input[:query_select]
+      end
+      influx_query = "#{query_select} from #{timeseries} \
+                      where time > #{start_date.to_i}s and time < #{end_date.to_i}s "
+      influx_query += query_input[:conditions] if query_input && query_input[:conditions]
```

Comment:
Apparently we should prefer `influx_query << query_input[:conditions] if ...` as is [faster and more Ruby-ish](https://github.com/bulletproofnetworks/ruby-style-guide#strings) :smile: